src/core/eval.ml

```diff
@@ -138,6 +144,14 @@ and eq_modulo : term -> term -> bool = fun a b ->
     | (Appl(t1,u1), Appl(t2,u2)) -> eq_modulo ((u1,u2)::(t1,t2)::l)
     | (Meta(m1,a1), Meta(m2,a2)) when m1 == m2 ->
         eq_modulo (if a1 == a2 then l else List.add_array2 a1 a2 l)
+    (* Try a δ conversion *)
```

Comment:
The case `| (Vari(x1)   , Vari(x2)   ) when Bindlib.eq_vars x1 x2 -> eq_modulo l` should be moved here.

Reply:
It should also be generalized as follows:

First, we need to change the output type of def_of so that it always returns a term option instead of raising Not_found.

```
| (Vari(x1) as a', Vari(x2) as b') ->
  if Bindlib.eq_vars x1 x2 then eq_modulo l
  else match def_of x1 ctx, def_of x2 ctx with
     | None, None -> raise Exit
     | Some t1, None -> eq_modulo ((t1,b')::l)
     | None, Some t2 -> eq_modulo ((a',t2)::l)
     | Some t1, Some t2 -> eq_modulo ((t1,t2)::l)
```

Reply:
Isn't it better to test equality of variables through `Bindlib` before searching in the context?

Reply:
right